### PR TITLE
Standardized country data API

### LINKED ISSUE #155: Implement standardized country flag data for mentor profiles
**Description:**
Update the backend to support standardized country selection and storage for mentor profiles.

**Tasks:**
- Update the database schema (if necessary)
- Implement API endpoint for retrieving the list of countries
- Implement server-side validation for country data (if necessary)

**Acceptance Criteria:**
- [ ] Create an API endpoint to serve a list of valid countries (e.g., /api/countries)
- [ ] Implement server-side validation to ensure only valid country codes are stored
- [ ] Update relevant queries to include country information where appropriate
- [ ] Ensure proper error handling for invalid country data

**Additional Information:**

**Related Dependencies or References:**
Related frontend issue: [https://github.com/sef-global/scholarx-frontend/issues/165](https://github.com/sef-global/scholarx-frontend/issues/165)

### PR DESCRIPTION
**Pull request by Disura-Randunu**
## Purpose

The purpose of this PR is to fix #155

## Goals

To implement a standardized country API endpoint to be used within the application

## Approach

Created a model, migration file & seeder. Added related service and route files which can be used in other parts of the application. JSON data file must be in the same directory as the seed file in build directory

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Disura-Randunu** [comment]
@anjula-sack Made the PR for countries API.

File: src/scripts/seed-db-countries.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import path from 'path'
+import fs from 'fs'
+import { dataSource } from '../configs/dbConfig'
+import { Country } from '../entities/country.entity'
+
+// Countries file must be in the same directory as the seeding file in build directory.
+const countriesDataFileName = 'countries.json'
+const countriesFilePath = path.join(__dirname, countriesDataFileName)
+
+export const seedCountries = async (): Promise<void> => {
```
Comment (anjula-sack) [inline src/scripts/seed-db-countries.ts:10]: As we discussed, let's move this script to the existing seed script @Disura-Randunu

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/scripts/seed-db-countries.ts (hunk 1 above)
Comment (Disura-Randunu) [reply src/scripts/seed-db-countries.ts:10]: sure. will update that

Comment (Disura-Randunu) [reply src/scripts/seed-db-countries.ts:10]: @anjula-sack btw, are we seeding the actual database through this script or manually inserting through the shell or something?

Comment (anjula-sack) [reply src/scripts/seed-db-countries.ts]: We will be running the command

Comment (Disura-Randunu) [reply src/scripts/seed-db-countries.ts]: @anjula-sack updated as disussed

**Review by anjula-sack — APPROVED** [review]
Thank you! @Disura-Randunu

**Pull request merged**